Formula/php.rb

```diff
@@ -0,0 +1,260 @@
+      end
+    end
+  end
+
+  def caveats
+    s = []
+
+    s << <<-EOS.undent
+      To enable PHP in Apache add the following to httpd.conf and restart Apache:
+          LoadModule php7_module #{HOMEBREW_PREFIX}/opt/php#{version.to_s[0..2].delete(".")}/libexec/apache2/libphp7.so
+
+          <FilesMatch \.php$>
+              SetHandler application/x-httpd-php
+          </FilesMatch>
+
+      Finally, check DirectoryIndex includes index.php
+          DirectoryIndex index.php index.html
+    EOS
+
+    s << <<-EOS.undent
+      The php.ini file can be found in:
+          #{etc}/php/#{php_version}/php.ini
+    EOS
+
+    s.join "\n"
+  end
+
+  def post_install
+    (var/"log").mkpath
+    touch var/"log/php-fpm.log"
+
+    chmod 0755, lib/"php/.channels"
+    chmod 0755, lib/"php/.channels/.alias"
+    chmod 0644, (Dir.glob(lib/"php/.channels/**/*", File::FNM_DOTMATCH).reject { |a| a =~ %r{\/\.{1,2}$} || File.directory?(a) })
+
+    %w[
+      php/.depdblock
+      php/.filemap
+      php/.depdb
+      php/.lock
+    ].each do |f|
+      chmod 0644, lib/f
+    end
+
+    php_ini = "#{etc}/php/#{php_version}/php.ini"
+    system bin/"pear", "config-set", "php_ini", php_ini, "system"
+
+    %w[
+      ldap
+      mcrypt
+      tidy
+      imap
+      opcache
+    ].each do |e|
+      config_path = (etc/"php/#{php_version}/conf.d/ext-#{e}.ini")
+      extension_type = (e == "opcache") ? "zend_extension" : "extension"
+      if config_path.exist?
+        inreplace config_path, %r{#{extension_type}=.*$}, "#{extension_type}=#{e}.so"
+      else
+        config_path.write <<-EOS.undent
+          [#{e}]
+          #{extension_type}="#{e}.so"
+        EOS
+      end
+    end
+  end
+
+  def php_version
+    version.to_s.split(".")[0..1].join(".")
+  end
+
+  plist_options :startup => true, :manual => "php-fpm --nodaemonize --fpm-config #{HOMEBREW_PREFIX}/etc/php/#{version.to_s.split(".")[0..1].join(".")}/php-fpm.conf"
+
+  def plist; <<-EOPLIST.undent
+    <?xml version="1.0" encoding="UTF-8"?>
+    <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+    <plist version="1.0">
+      <dict>
+        <key>KeepAlive</key>
+        <true/>
+        <key>Label</key>
+        <string>#{plist_name}</string>
+        <key>ProgramArguments</key>
+        <array>
+          <string>#{opt_sbin}/php-fpm</string>
+          <string>--nodaemonize</string>
+          <string>--fpm-config</string>
+          <string>#{etc}/php/#{php_version}/php-fpm.conf</string>
+        </array>
+        <key>RunAtLoad</key>
+        <true/>
+        <key>WorkingDirectory</key>
+        <string>#{var}</string>
+        <key>StandardErrorPath</key>
+        <string>#{opt_prefix}/var/log/php-fpm.log</string>
+      </dict>
+    </plist>
+    EOPLIST
+  end
+
+  test do
+    system "#{bin}/php", "-i"
+    system "#{sbin}/php-fpm", "-y", "#{etc}/php/#{php_version}/php-fpm.conf", "-t"
```

Comment:
Same as above, drop the `-y` option. These tests can fail based on user changed configuration, just an FYI.

It might also be worth validating that all the built shared modules are actually working by ensuring that they appear in the output of `#{bin}/php -m`.

It might also be worthwhile to test the other sapis:
* apache2handler: run `#{Formula['httpd24'].bin}/httpd -M` and look for `php7_module` in the output. The command may require the module to be forcibly loaded via option `-C "LoadModule php7_module #{opt_libexec}/apache2/libphp7.so"`
* phpdgb: run `#{bin}/phpdbg -V` 
* cgi: run `#{bin}/php-cgi -m`

Reply:
The tests are mainly for the CI so it shouldn't really matter if they fail on a users system imo. I'll add the other tests.